phy/mod_difest.F

```diff
@@ -1161,7 +1162,7 @@ subroutine difest_vertical_hyb(m,n,mm,nn,k1m,k1n)
      .                               cellHeight,OBLdepth(i,j))
 
              ! gets index of the level and interface above hbl
-             kOBL = CVMix_kpp_compute_kOBL_depth(iFaceHeight,
+             kOBL(i,j) = CVMix_kpp_compute_kOBL_depth(iFaceHeight,
```

Comment:
I am a bit uncertain, but `CVMix_kpp_compute_kopb_depth()` i) seem to return a real value and ii) depending on which index one wants, it potentially requires to subtract 1 (not sure about the vertical grid layout in BLOM) iii) why calculating it again - and not using hOBL? - from CVMix:
```
! !DESCRIPTION:
!  Computes the index of the level and interface above OBL\_depth. The index is
!  stored as a real number, and the integer index can be solved for in the
!  following way:\\
!    \verb|kt| = index of cell center above OBL\_depth = \verb|nint(kOBL_depth)-1|
!    \verb|kw| = index of interface above OBL\_depth = \verb|floor(kOBL_depth)|
```
In any case, to be clean, I guess it would be good to explicitly convert from real to integer for the k-index. This applies again below.

Reply:
Thanks @jmaerz ! 

@matsbn @milicak I would like to hear from you about this issue.

I did not look into the CVMix routine itself when I made these code changes, but I agree that this looks suspicious. For iHAMOCC we probably want the index of the cell center above `OBL_depth`, not the index of the interface. 

`hOBL` is computed twice, first by the subroutine `CVMix_kpp_compute_OBL_depth` and later by function `CVMix_kpp_compute_kOBL_depth`. The function `CVMix_kpp_compute_kOBL_depth` is used internally in `CVMix_kpp_compute_OBL_depth`, so there should be no need to do this again. Then `kOBL` is computed (twice) by the function `CVMix_kpp_compute_kOBL_depth`.

`kOBL` is used in the calculation of buoyancy flux acting on the OBL, so in this case the index may need to reference the interface. If so, we may want to define a new index `OBL_cell_center = nint(hOBL) - 1` for use in iHAMOCC.